pkg/util/resource/resource.go

```diff
@@ -103,3 +107,32 @@ func QuantityToFloat(q *resource.Quantity) float64 {
 	}
 	return float64(q.MilliValue()) / 1000
 }
+
+func MulResources(res map[corev1.ResourceName]resource.Quantity, multiplier float64) map[corev1.ResourceName]resource.Quantity {
+	result := make(map[corev1.ResourceName]resource.Quantity, len(res))
+	for k, v := range res {
+		sum := float64(resources.ResourceValue(k, v))
+		sum *= multiplier
+		result[k] = resources.ResourceQuantity(k, int64(sum))
+	}
+	return result
+}
+
+func AddResources(a, b map[corev1.ResourceName]resource.Quantity) map[corev1.ResourceName]resource.Quantity {
```

Comment:
I think `AddResources` and `MulResources` can be simplified (and reduce risk of bugs) by reusing the `Requests` type.

Consider: define a new constructor for `Requests`, `FromResourceList(corev1.ResourceList) Requests`. Then use respective Add/Mul functions. Then convert back with already defined `ToResourceList`

Reply:
Changed the types to simplify the logic and use some of the existing functions, ptal